feature/enhance-url-pattern-matching

alvinjohnsonso opened the pull request:
- Added tawk/url-utils lib for the updated pattern matching.
- Updated pattern matching logic to comply with the tawk/url-utils lib requirements.
- Updated visibility options UI to show the users the new patterns available.
- Created upgrade script to update the existing patterns based on the tawk/url-utils lib pattern matching.

alvinjohnsonso committed:
added tawk-url-utils package

alvinjohnsonso committed:
added release github workflow

alvinjohnsonso committed:
updated tawk-url-utils usage for url matching

alvinjohnsonso committed:
updated release workflow to build and upload artifact on publish of release

alvinjohnsonso committed:
Merge branch 'master' of https://github.com/tawk/tawk-wordpress into feature/enhance-url-pattern-matching

alvinjohnsonso committed:
Merge branch 'r-0.7.0' of github.com:alvinjohnsonso/tawk-wordpress into feature/enhance-url-pattern-matching

alvinjohnsonso committed:
removed post-install-cmd script due to installed dev dependencies already adding the installed_paths for the sniffs

alvinjohnsonso committed:
Merge pull request #11 from alvinjohnsonso/fix/phpcs-sniffs-installed-path

fix phpcs sniffs installed path

alvinjohnsonso committed:
Merge branch 'r-0.7.0' of github.com:alvinjohnsonso/tawk-wordpress into feature/enhance-url-pattern-matching

alvinjohnsonso committed:
- added composer build scripts for dev and prod dependencies
- integrated url and path matching module from tawk-url-utils lib

alvinjohnsonso committed:
updated creating release artifact workflow

alvinjohnsonso committed:
updated tawk-url-utils

alvinjohnsonso committed:
updated github create release artifact workflow to rename the folder within the zip file

alvinjohnsonso committed:
Added wildcard valid/invalid formats info

alvinjohnsonso committed:
added upgrade script for version upgrades

alvinjohnsonso committed:
allow upgrade hook to only be registered on admin area

alvinjohnsonso committed:
removed unused variable

alvinjohnsonso committed:
- Removed js functions that hides and disables the other switches on visibility option tab
- Added URL exclusion section. Including the accepted patterns as a tooltip.

alvinjohnsonso committed:
Removed upgrade script from main plugin file; Added upgrade manager to handle upgrades;

alvinjohnsonso committed:
updated tawk-url-utils lib

alvinjohnsonso committed:
Merge branch 'feature/enhance-url-pattern-matching' of github.com:alvinjohnsonso/tawk-wordpress into visibility-options-ui-update

alvinjohnsonso committed:
Merge branch 'feature/enhance-url-pattern-matching' of github.com:alvinjohnsonso/tawk-wordpress into db-upgrade-script

alvinjohnsonso committed:
made the visibility options to have uniform letter casing

alvinjohnsonso committed:
Merge branch 'release/pattern-matching-and-automated-tests' of https://github.com/tawk/tawk-wordpress into feature/enhance-url-pattern-matching

alvinjohnsonso committed:
Merge pull request #24 from alvinjohnsonso/visibility-options-ui-update

Visibility options UI update

alvinjohnsonso committed:
Merge pull request #25 from alvinjohnsonso/db-upgrade-script

Created DB upgrade script to add another record of the patterns with wildcard without the wildcard.

alvinjohnsonso committed:
Retrieved version from file description

alvinjohnsonso committed:
fixed included url list are not adding the new pattern without wildcard; skipped empty url

alvinjohnsonso committed:
reverted back show/hide visibility options js script

alvinjohnsonso committed:
Merge pull request #26 from alvinjohnsonso/fix/plugin-version-ref-and-include-url-pattern-migrate

- retrieved plugin version from plugin file data
- fixed migration script for include URL list

GeekOfAges commented on tawkto/upgrade.manager.php:
```diff
@@ -0,0 +1,92 @@
+		}
+
+		// update stored plugin version for next time.
+		update_option( $this->version_var_name, $this->curr_ver );
```
This needs to happen after every version upgrade applied to avoid re-applying migrations already applied and an error is encountered down-stream.

alvinjohnsonso replied:
Added the fix on this PR https://github.com/alvinjohnsonso/tawk-wordpress/pull/27/files

GeekOfAges commented on tawkto/upgrade.manager.php:
```diff
@@ -0,0 +1,92 @@
+		// All plugins prior to the current version needs the upgrade.
+		if ( version_compare( $this->prev_ver, $this->curr_ver ) < 0 ) {
+			// are there upgrade steps depending on how out-of-date?
+			foreach ( $this->upgrades as $next_ver ) {
```
This will apply all upgrades each time migration is ran

alvinjohnsonso replied:
Added the fix on this PR Added the fix on this PR https://github.com/alvinjohnsonso/tawk-wordpress/pull/27/files

GeekOfAges commented on tawkto/upgrades/version.070.php:
```diff
@@ -0,0 +1,76 @@
+		$new_included_url_list = array();
+		$new_excluded_url_list = array();
+
+		foreach ( $included_url_list as $included_url ) {
```
Could be extracted to function since same thing is applied to 2 lists

alvinjohnsonso replied:
Added the fix on this PR Added the fix on this PR https://github.com/alvinjohnsonso/tawk-wordpress/pull/27/files

GeekOfAges commented on tawkto/upgrades/version.070.php:
```diff
@@ -0,0 +1,76 @@
+				continue;
+			}
+
+			$new_included_url_list[] = $new_included_url;
```
List should be unique. We may have added a URL that already exists in list

alvinjohnsonso replied:
Added the fix on this PR Added the fix on this PR https://github.com/alvinjohnsonso/tawk-wordpress/pull/27/files

alvinjohnsonso committed:
- created getter function to retrieve plugin version
- created process_patterns function on the upgrade script to remove repeated logic on doing the upgrade process
- checked for added patterns to the new list to have a unique list

alvinjohnsonso committed:
Merge pull request #27 from alvinjohnsonso/fix/upgrade-script-multiple-execution-and-unique-list

fix/upgrade-script-multiple-execution-and-unique-list
